Source/WebCore/platform/graphics/cg/ColorCG.cpp

```diff
@@ -167,7 +167,14 @@ static RetainPtr<CGColorRef> createCGColor(const Color& color)
 {
     auto [colorSpace, components] = color.colorSpaceAndResolvedColorComponents();
     auto [cgColorSpace, cgCompatibleComponents] = convertToCGCompatibleComponents(colorSpace, components);
-    
+
+    // Clamp the extended colors so they look the same on SDR and HDR layers.
+    if (CGColorSpaceUsesExtendedRange(cgColorSpace)) {
+        auto [c1, c2, c3, c4] = cgCompatibleComponents;
+        auto clampedComponents = makeFromComponentsClamping<SRGBA<float>>(c1, c2, c3, c4);
+        cgCompatibleComponents = asColorComponents(clampedComponents.resolved());
+    }
```

Comment:
This will break any use of greater than sRGB color spaces, including non-"HDR" ones. 

If you are going to introduce clamping here, it will need to be clamped to the SDR output colorspace. That is, the output colorspace that would have been used had no HDR content been included in the page.